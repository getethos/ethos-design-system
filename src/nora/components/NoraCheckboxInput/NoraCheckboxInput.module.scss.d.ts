// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'NoraCheckboxInput': string;
}
export const cssExports: CssExports;
export default cssExports;
