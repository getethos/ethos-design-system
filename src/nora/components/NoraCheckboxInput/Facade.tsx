import * as React from 'react'
type FacadeProps = {
  classes: string
}
// Nora's Facade (checkbox shape) needs to be smaller to
// account for overall smaller font sizes we use in Nora
export const Facade: React.FC<FacadeProps> = ({ classes }) => {
  return (
    <svg
      className={classes}
      width="18"
      height="18"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill="#fff"
        d="M 5.25 9.433594 L 2.816406 7 L 1.988281 7.820312 L 5.25 11.082031 L 12.25 4.082031 L 11.429688 3.261719 Z M 5.25 9.433594 "
      />
    </svg>
  )
}
