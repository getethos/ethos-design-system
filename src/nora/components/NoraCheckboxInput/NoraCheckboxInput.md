`NoraCheckboxInput` is a simple wrapper component over `CheckboxInput`
that accounts for the generally smaller font sizes we use in Nora.

```jsx
<NoraCheckboxInput
  name="le-check-unchecked"
  data-tid="le-tid-unchecked"
  validator={(n) => {
    console.log('yo yo')
    if (n === true) {
      return ''
    }
    return 'You must agree to submit form'
  }}
>
  I agree to the{' '}
  <a href="/" target="_blank">
    Agreement
  </a>{' '}
  , and the{' '}
  <a href="/" target="_blank">
    Other Agreement
  </a>
</NoraCheckboxInput>
```

Example with one checkbox setting the value of another

```jsx
import React, { useState } from 'react'

function NoraCheckboxExample() {

  const [labSelected, setLabSelected] = useState(true);
  const [ceaSelected, setCeaSelected] = useState(false);

  // Instead of a validator setting the state of the labs, we can also consider disabling the checkbox if one of the children are selected
  return (
    <>
      <NoraCheckboxInput
        name="labs"
        data-tid="labs-checkbox"
        initialValue={labSelected}
        validator={(val) => {
          if (val === false && ceaSelected === true) {
            setLabSelected(true);
          }
          return ''
        }}
      >
        Labs
        <NoraCheckboxInput
          name="cea"
          data-tid="cea-checkbox"
          initialValue={ceaSelected}
          validator={(val) => {
            setCeaSelected(val);
            if (val === true) {
              setLabSelected(true)
            }
            return ''
          }}
        >
          CEA
        </NoraCheckboxInput>
      </NoraCheckboxInput>
    </>
  )
}

<NoraCheckboxExample />
```
