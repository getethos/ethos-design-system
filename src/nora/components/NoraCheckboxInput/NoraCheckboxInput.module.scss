@import '../Attributes.css';

.NoraCheckboxInput [type='checkbox'] ~ svg {
  border-radius: var(--Radius-3);
  width: 14px;
  height: 14px;
}

.NoraCheckboxInput > label > div {
  top: 3px;
}

.NoraCheckboxInput [type='checkbox']:checked ~ svg {
  background-color: var(--NoraGreen-400);
}
