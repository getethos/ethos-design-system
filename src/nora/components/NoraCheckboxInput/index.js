export { NoraCheckboxInput } from './NoraCheckboxInput.js'
