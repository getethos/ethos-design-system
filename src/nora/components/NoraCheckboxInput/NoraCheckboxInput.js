import React from 'react'
import PropTypes from 'prop-types'
import { CheckboxInput } from '../../../components/index'
import { Facade } from './Facade.js'
import styles from './NoraCheckboxInput.module.scss'

export const NoraCheckboxInput = ({
  validator,
  children,
  name,
  initialValue,
  ...rest
}) => {
  const renderFacade = (klasses) => <Facade classes={klasses} />
  return (
    <div className={styles.NoraCheckboxInput}>
      <CheckboxInput
        initialValue={initialValue}
        name={name}
        data-tid={rest['data-tid']}
        validator={validator}
        facadeRenderer={renderFacade}
        {...rest}
      >
        {children}
      </CheckboxInput>
    </div>
  )
}

NoraCheckboxInput.propTypes = {
  name: PropTypes.string.isRequired,
  initialValue: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  'data-tid': PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  validator: PropTypes.func,
}
