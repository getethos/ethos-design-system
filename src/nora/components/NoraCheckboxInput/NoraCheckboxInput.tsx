import * as React from 'react'
import { PublicFormProps } from '../../../components/Form/Form'
import { CheckboxInput } from '../../../components/index'
import { Facade } from './Facade'
import styles from './NoraCheckboxInput.module.scss'
type NoraCheckboxInputProps = PublicFormProps & {
  initialValue?: string | boolean
  checked?: boolean
  disabled?: boolean
}
export const NoraCheckboxInput: React.FC<NoraCheckboxInputProps> = ({
  validator,
  children,
  name,
  initialValue,
  checked,
  disabled,
  ...rest
}) => {
  const renderFacade = (klasses) => <Facade classes={klasses} />
  return (
    <div className={styles.NoraCheckboxInput}>
      <CheckboxInput
        initialValue={initialValue}
        name={name}
        data-tid={rest['data-tid']}
        validator={validator}
        facadeRenderer={renderFacade}
        disabled={disabled}
        checked={checked}
        {...rest}
      >
        {children}
      </CheckboxInput>
    </div>
  )
}
