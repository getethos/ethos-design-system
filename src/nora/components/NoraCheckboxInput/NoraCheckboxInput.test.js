import React from 'react'
import { NoraCheckboxInput } from './NoraCheckboxInput'
import renderer from 'react-test-renderer'

const formChangeHandlerStub = jest.fn()

describe('NoraCheckboxInput', () => {
  it('default rendering', () => {
    const tree = renderer
      .create(
        <NoraCheckboxInput
          name="le-check"
          data-tid="le-tid"
          formChangeHandler={formChangeHandlerStub}
          validator={(x) => (x === true ? '' : 'Nope!')}
        >
          I agree to the{' '}
          <a href="/" target="_blank">
            Agreement
          </a>
        </NoraCheckboxInput>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
