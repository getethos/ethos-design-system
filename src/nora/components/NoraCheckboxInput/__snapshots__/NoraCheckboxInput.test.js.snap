// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoraCheckboxInput default rendering 1`] = `
<div
  className="NoraCheckboxInput"
>
  <label
    className="root"
    htmlFor="le-check"
  >
    <div
      className="checkboxWrapper"
    >
      <input
        checked={false}
        className="CheckboxInput"
        data-tid="le-tid"
        id="le-check"
        name="le-check"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="checkbox"
      />
      <svg
        className="Facade FacadeDefaultBackground FacadeBorder"
        fill="none"
        height="18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 5.25 9.433594 L 2.816406 7 L 1.988281 7.820312 L 5.25 11.082031 L 12.25 4.082031 L 11.429688 3.261719 Z M 5.25 9.433594 "
          fill="#fff"
        />
      </svg>
    </div>
    <div
      className="Body Theinhardt Regular400 GrayPrimary"
    >
      I agree to the
       
      <a
        href="/"
        target="_blank"
      >
        Agreement
      </a>
    </div>
  </label>
  
</div>
`;
